client/components/pagination/index.jsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import PaginationPage from './pagination-page';
+
+class Pagination extends Component {
+	static propTypes = {
+		page: React.PropTypes.number.isRequired,
+		perPage: React.PropTypes.number.isRequired,
+		total: React.PropTypes.number,
+		pageClick: React.PropTypes.func.isRequired,
+	}
+
+	render() {
+		const { page, perPage, total } = this.props;
+		let pageList = [];
+		const pageCount = Math.ceil( total / perPage );
+
+		if ( pageCount > 1 ) {
+			pageList = [
+				1,
+				page - 2,
+				page - 1,
+				page,
+				page + 1,
+				page + 2,
+				pageCount,
+			];
+			pageList.sort( function( a, b ) {
+				return a - b;
+			} );
+
+			// Remove pages less than 1, or greater than total number of pages, and remove duplicates
+			pageList = pageList.filter( function( pageNumber, index, originalPageList ) {
+				return (
+					pageNumber >= 1 &&
+					pageNumber <= pageCount &&
+					originalPageList.lastIndexOf( pageNumber ) === index
+				);
+			} );
```

Comment:
Since we're going full ES6, we could refactor this as an arrow function:

```es6
pageList = pageList.filter( ( pageNumber, index, originalPageList ) => { ... } );
```

I'd keep the explicit return though, as it improves the readability.